frontend/src/metabase-lib/queries/utils/dataset.ts

```diff
@@ -52,35 +52,46 @@ export function findColumnIndexForColumnSetting(
       return columnIndex;
     }
   }
+
   // if that fails, find by column name
   return _.findIndex(columns, col => col.name === columnSetting.name);
 }
 
-export function findColumnSettingIndexForColumn(
+type FieldRefWithIndex = {
+  fieldRef: FieldReference;
+  originalIndex: number;
+};
+
+export function findColumnSettingIndexesForColumns(
   query: Lib.Query,
+  stageIndex: number,
+  columns: DatasetColumn[],
   columnSettings: TableColumnOrderSetting[],
-  column: DatasetColumn,
 ) {
-  // ignore settings without fieldRef but preserve indexes
-  const items = columnSettings.flatMap((item, settingIndex) => {
-    const fieldRef = normalize(item.fieldRef);
-    return fieldRef ? [{ fieldRef, settingIndex }] : [];
-  });
+  const fieldRefs = columnSettings.reduce(
+    (fieldRefs: FieldRefWithIndex[], { fieldRef }, originalIndex) => {
+      if (fieldRef != null) {
```

Comment:
`normalize` should not be needed or there is a bug
Previously invalid field refs were created for column settings without them, and that would lead to exceptions in `findColumnIndexesFromLegacyRefs`